Project Solana Address Support

aminlatifi opened the pull request:
#1187

aminlatifi committed:
Merge pull request #1193 from Giveth/feat/solana_support

Solana user profile support

aminlatifi committed:
Initial add solana address support to project address

aminlatifi committed:
Added chaintype to graphql validator and project verification form

aminlatifi committed:
Added chainType to token
Refactored project filter query builder
Added tests

aminlatifi committed:
Fixed issue in test graphql query
Added some tests

aminlatifi committed:
Fixed issues with tests

CarlosQ96 commented on src/resolvers/projectResolver.ts:
```diff
@@ -475,140 +482,84 @@ export class ProjectResolver {
+      }
+    });
+
+    if (networkIds.length > 0 || acceptFundOnSolanaSeen) {
```
We will have to limit it in the frontend. Like you said both are not posible

aminlatifi replied:
No, I believe we don't need to limit the front end. Before this, we had an `AND` conjunction between accepting funds on chains, so choosing multiple ones had no meaning. After this, we will have an `OR` conjunction between them.

CarlosQ96 reviewed (approved):
Looks great functionality wise, I will add the tokens in my feature branch for the donation.

aminlatifi commented on src/resolvers/projectResolver.test.ts:
```diff
@@ -1262,6 +1277,145 @@ function allProjectsTestCases() {
+    const result = await axios.post(graphqlUrl, {
+      query: fetchMultiFilterAllProjectsQuery,
+      variables: {
+        filters: ['AcceptFundOnMainnet', 'AcceptFundOnSolana'],
```
@CarlosQ96 Look at this test, all other test scenarios we have had before only test filters with single critrion.

mohammadranjbarz reviewed (approved):
Thanks @aminlatifi  for implementing Solana and refactoring some old codes.
LGTM, but I think it's better to add `chainType` to admin bro then admins can filter donations/tokens based on chainType

https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/tokenTab.ts#L174

https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/donationTab.ts#L372

aminlatifi committed:
Feat/solana project address staging merged (#1199)

* Added fill snapshot round number back

* Removed blockNumber from snapshot in tests

* Added fill snapshot round number

* Fixed migrations to be compatible with removing blockNumber from history

* Improve networks filters in all projects query

related to #1096

* improve calculations priority queue

* upgrade master test env for new auth service server

* add tests for staging goerli and alfajores

* Fix test cases for alfadores network

related to #1096

* Fix test cases for alfadores network

related to #1096

* Update staging-pipeline.yml

* 1.16.0

* Changed balance aggregator query to get

* Fixed param passing to getLatestBalances of power balance aggregator

* 1.16.1

* Return list of active campaigns in projectBySlug web service

related to #1051

* rewrite totalReceived query

* Update totalReceived of project owners correctly after verifying donations

https://github.com/Giveth/giveth-dapps-v2/issues/3021

* Update totalReceived of project owners correctly after verifying donations

https://github.com/Giveth/giveth-dapps-v2/issues/3021

* Delete test user after some test cases

* Fix Auth microservice address in config

* Add cache to qfRound estimated matching queries

related to #1103

* removed unused code

* Add QF_ROUND_ESTIMATED_MATCHING_CACHE_DURATION to example.env

* add eligible donations to qfround entity

* add tests for qfround eligible donations

* add adminjs eligiblenetworks editable on qfround

* Update staging-pipeline.yml

Added staging DB to testing in the CI

* add labels to the networks in adminjs for qfround

* add migration to fill in previous networkIds in qfrounds

* fix eligible donations migration

* fix migration table names for qfround networks

* fix migration file for eligible networks

* fix qfround fill networks migration

* add query for knowing if wallet was used

* add query for knowing if wallet was used

* improve query for user related address

* improve query for user related address

* unify project counts in userByAddress and projectsByUserId

* unify project counts in userByAddress and projectsByUserId

* improve condition for projectCount

* fix projectCount query

* improve condition for projectCount

* fix projectCount query

* Add logs for checkProjectVerificationStatus to can investigate better

related to https://github.com/Giveth/impact-graph/issues/932#issuecomment-1733733608

* Add Other types of campaigns to projectBySlug webservice

https://github.com/Giveth/impact-graph/issues/1051#issuecomment-1733713799

* Fix calling remind project update notification and more logs

* Add more logs

* Change name of fetchProjectsBySlugQuery field

* add qfFilter with estimatedMatchingView

* add sorting by qfround raised funds

* Implement doesDonatedToProjectInQfRound webservice

related #1138

* Move caching project campaign slugs process to a separated worker

related #1051

* fix relationships for estimated matching filters

* Add ETC network

* Put campaign projects cache in redis instead of saving that in memory

* Change default value of CACHE_PROJECT_CAMPAIGNS_CRONJOB_EXPRESSION

* change to leftJoin on estimatedMatchingTable

* fix tests

* modify qfround filter join

* order null donations to last position

* add optimism data queries

* add graphql queries examples

* Add some logs

* Change default value of CACHE_PROJECT_CAMPAIGNS_CRONJOB_EXPRESSION

* Put calling updateInstantPowerBalances() in try..catch...

* Add comment

* 1.17.0

* Get giv price from givback-calculation instead of monoswap

* Add GIVETH_GIV_PRICES_URL in test.env

* Add some GLO and pyUSD stable coins

related to https://github.com/Giveth/giveth-dapps-v2/issues/3221

* Call cacheProjectCampaigns() in main thread

* Fix filling xdai donation price

* Add octant donations to DB

* Fix test cases

* add idriss donation logic and cronjob

* Integrate with Mordor ETC testnet chain

related #1157, #1156

* Modify Octant donation migration file

* Fix migration file names for stable coins

* Fix build problem

* add creation of donation and user from idriss donation

* Fix donation resolver test case for etc testnet

* fix cache and improve optimism queries

* fix donations per category query for optimism

* Change date of Public Noun donation

* Add edit button for donation tab in admin panel

related to #1161

* improve query to count for optimism

* Add transactionId to filter in donation tab in admin panel

* simplify donation total for optimism

* fix cache key for analytics dashboard queries

* add optimism data queries

* add graphql queries examples

* fix cache and improve optimism queries

* fix donations per category query for optimism

* improve query to count for optimism

* simplify donation total for optimism

* fix cache key for analytics dashboard queries

* Add price adapters

* Fix editing donations tab

* add additional filters for optimism queries

* add eligible recipients for idriss donations loop

* fix cache for new filters

* add new cache with new filter params

* add additional filters for optimism queries

* fix cache for new filters

* add new cache with new filter params

* Get ETC prices for ETC and Mordor network tokens

related to #1156, #1157, #1158

* Add logs in price adapters

* finishing changes to idriss donation creation

* Fix Adding addresses to managingFunds

* Fix editing donations in adminjs panel

* fix date in idriss donation

* improve idriss integration

* add error catching for idriss

* fix monoswap prices for idriss donations

* 1.18.0

* Change migration name of AddExternalDonationsFields

* Rename migration files to fix the orders

* Rename migration files to fix the orders

* Comment unstable test case

* add subsquid subgraph to optimize idriss integration

* change idriss cronjob time

* add idriss donation export in adminJs as button

* idriss improvements

* fix idriss import queries

* fix project status fetching for idriss

* add project status to getVerificationFormByProjectId query

* add project status to getVerificationFormByProjectId query

* Add sent matching fund to total donations of project and total received of users

related to https://github.com/Giveth/giveth-dapps-v2/issues/3386

* Fix test case for adding matching fund to toal donation amount of projects

* Add knownAsSybilAddress to user entity and adding export sybils data to admin panel

* Move get qfRound data buttons to qfRound tab

* Integrate googlesheet stuff with sybil analysis

* Fix types of QfRoundDonationRow

* Add some comments

* add project_actual_matching_view table

* Fix migraiton file for actual matching materialized views

* add qfRound Spreadhsheet logic and actualmatching

* add qfRound Spreadhsheet logic and actualmatching

* Fix emptyness of googlesheet

* Implement qfRoundStats webservice

related to https://github.com/Giveth/giveth-dapps-v2/issues/3250

* Change allProjects webservice to support qfRoundSlug

related to https://github.com/Giveth/giveth-dapps-v2/issues/3250

* Fix test cases to create qfRound

* Fix test cases to create qfRound

* Fix test cases to create qfRound

* Fix test cases to create qfRound

* Fix updateProjectWithVerificationForm

* Revert updateUserTotalReceived() funciton

* Fix updateUserTotalReceived() test case

* Fix updateUserTotalReceived() test case

* Fix mordor testnet node url

* Read node rpc urls from process.env instead of config

* Add log

* Fix etc provider url in github actions config

* Fix AddSlugToQfRound migration

* Insert donations to db based on distributed funds for qfRound projects

related to #1186

* Empty commit to trigger CI/CD

* Fix query to just add donation for qfRound histories that have full data

* Fix typo error

* Fix insertDonationsFromQfRoundHistory query

* Move ens address to constant parameter

* Fix integration tests for creating donations with matchingFund data

* Fix integration tests for creating donations with matchingFund data

* Fix integration tests for creating donations with matchingFund data

* Remove unused tests

* add safeTransaction to donation logic

* fix uniqueness index on donation table

* comment a network tests

* fix nullability of transactionId

* make nonce nullable for safe donations

* add nullability to nonce in createdonationResolver

* remove undefined clause from joi validations

* Fix insertDonationsFromQfRoundHistory when can not find corosponding address

* Call refreshProjectDonationSummaryView() immediately after insertDonationsFromQfRoundHistory)

* add validations for multisig donations

* add abis decoder for multisig transactions and parsers

* fetch from and to addresses correctly for multisig validations

* fix native token transfer for multisig transactions

* fix transaction amount in tests for multisig

* fix fromAddress edgecase for multisig

* make params options for transactionData

* Revert unwanted changes happend after merge

* Fixed issues happend after merge

* Fixed an issue in setting chainType in projectUpdate

---------

Co-authored-by: Mohammad Ranjbar Z <mranjbar.z2993@gmail.com>
Co-authored-by: Carlos <carlos.quintero096@gmail.com>
Co-authored-by: Moe Shehab <52987806+mhmdksh@users.noreply.github.com>
Co-authored-by: CarlosQ96 <92376054+CarlosQ96@users.noreply.github.com>
Co-authored-by: Krati Jain <kratijain88@gmail.com>
Co-authored-by: Mateo Daza <mateodaza@gmail.com>

aminlatifi commented:
Closed and made #1200

aminlatifi commented:
> Thanks @aminlatifi for implementing Solana and refactoring some old codes. LGTM, but I think it's better to add `chainType` to admin bro then admins can filter donations/tokens based on chainType
> 
> https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/tokenTab.ts#L174
> 
> https://github.com/Giveth/impact-graph/blob/248356696da9a949101d3dd4e911b33159c97154/src/server/adminJs/tabs/donationTab.ts#L372

Let's solve it as a separate issue, because it's not clear for me how it would be!
